Analysis/Tasks/PWGHF/HFCandidateCreator2Prong.cxx

```diff
@@ -69,8 +69,8 @@ struct HFCandidateCreator2Prong {
 
     // loop over pairs of track indices
     for (const auto& rowTrackIndexProng2 : rowsTrackIndexProng2) {
-      auto trackParVarPos1 = getTrackParCov(rowTrackIndexProng2.index0());
-      auto trackParVarNeg1 = getTrackParCov(rowTrackIndexProng2.index1());
+      auto trackParVarPos1 = getTrackParCov(rowTrackIndexProng2.index0_as<aod::FullTracks>());
+      auto trackParVarNeg1 = getTrackParCov(rowTrackIndexProng2.index1_as<aod::FullTracks>());
```

Comment:
`FullTracks` -> `BigTracks`

Reply:
Thanks, Vit. For retrieving the covariance matrix FullTracks should be enough. Or is the getTrackParCov accessing members of BigTracks?

Reply:
> Thanks, Vit. For retrieving the covariance matrix FullTracks should be enough. Or is the getTrackParCov accessing members of BigTracks?

No, `getTrackParCov` does not need `BigTracks`. The point is that the task subscribes to `BigTracks`, so casting `Tracks` to `FullTracks` does not work, as far as I know.

Reply:
Or is casting to a subset of a subscribed type supposed to work?

Reply:
I see. Good question. @aalkin ?

Reply:
Casting to a subset should work, but I would avoid that just in case.

Reply:
Hello, 

In case it is useful, here my experience. 

Thinking that it would work, I did what I think you mean here: I subscribed to a quite large table, and then casted to a subset of it:
```
using BigTracks = soa::Join<aod::Tracks, aod::TracksCov, aod::TracksExtra, aod::TracksExtended>;
using BigTracksMC = soa::Join<BigTracks, aod::McTrackLabels>;
...
  void process(aod::Collision const& collision, aod::V0s const& V0s,
               BigTracksMC const& tracks, aod::McParticles const& particlesMC)
{
    for (auto& V0 : V0s) {
...
     if (V0.posTrack_as<BigTracks>().tpcNClsCrossedRows() < mincrossedrows) {
	if (labelPos == 981525 || labelPos == 1080007 || labelPos == 1151717 || labelPos == 46082 || labelPos == 386994 || labelPos == 1178152 || labelPos == 1386973) {
	  LOG(INFO) << "posTrack has " << V0.posTrack_as<BigTracksMC>().tpcNClsCrossedRows() << " crossed rows, cut at " << mincrossedrows;
	}
        continue;
      }
```

but I got a crash:
```
[10180:lf-lambdakzeroprefilterpairs]:  *** Break *** segmentation violation
[10180:lf-lambdakzeroprefilterpairs]: aprocessor_id=lf-lambdakzeroprefilterpairs
...
[11445:lf-lambdakzeroprefilterpairs]: #7  std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count (__r=..., this=<optimized out>) at /usr/include/c++/9/bits/shared_ptr_base.h:1167
[11445:lf-lambdakzeroprefilterpairs]: #8  std::__shared_ptr<arrow::Array, (__gnu_cxx::_Lock_policy)2>::__shared_ptr (this=<synthetic pointer>) at /usr/include/c++/9/bits/shared_ptr_base.h:1167
[11445:lf-lambdakzeroprefilterpairs]: #9  std::shared_ptr<arrow::Array>::shared_ptr (this=<synthetic pointer>) at /usr/include/c++/9/bits/shared_ptr.h:129
[11445:lf-lambdakzeroprefilterpairs]: #10 arrow::ChunkedArray::chunk (i=32552, this=0x7fffce6d4c00) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/ubuntu2004_x86-64/arrow/v1.0.0-1/include/arrow/chunked_array.h:97
[11445:lf-lambdakzeroprefilterpairs]: #11 o2::soa::ColumnIterator<signed char, o2::soa::Chunked>::getCurrentArray (this=0x7fffce6d0f20, this=0x7fffce6d0f20) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/ASoA.h:264
[11445:lf-lambdakzeroprefilterpairs]: #12 o2::soa::ColumnIterator<signed char, o2::soa::Chunked>::nextChunk (this=0x7fffce6d0f20) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/ASoA.h:169
[11445:lf-lambdakzeroprefilterpairs]: #13 o2::soa::ColumnIterator<signed char, o2::soa::Chunked>::operator* (this=0x7fffce6d0f20) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/ASoA.h:216
[11445:lf-lambdakzeroprefilterpairs]: #14 o2::aod::track::TPCNClsCrossedRows<o2::aod::track::TPCNClsFindable, o2::aod::track::TPCNClsFindableMinusCrossedRows>::boundGetter<0ul, 1ul> (this=0x7fffce6d1110) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/AnalysisDataModel.h:177
[11445:lf-lambdakzeroprefilterpairs]: #15 0x000055e4961b6891 in o2::aod::track::TPCNClsCrossedRows<o2::aod::track::TPCNClsFindable, o2::aod::track::TPCNClsFindableMinusCrossedRows>::tpcNClsCrossedRows<>() const (this=0x7fffce6d1110) at /home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Framework/Core/include/Framework/AnalysisDataModel.h:177
```
(`TPCNClsCrossedRows` is in `aod::TracksExtra`), so it seems that using a sub-table of the one you subscribe to does not work. Everything is fine if I cast to the "full" table.

Chiara

Reply:
Thanks! Changed.